### PR TITLE
slib 3b4 (new formula)

### PR DESCRIPTION
**Pull request by kmizumar**
New formula for slib - The SLIB Portable Scheme Library.

### CODE REVIEW

File: Library/Formula/slib.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    system "make", "install", "INSTALL_INFO=install-info"
+  end
+
+  def caveats; <<-EOS.undent
+    SLIB distribution is installed at /usr/local/lib/slib/.
+
+    See `info slib' to see how to configure your Scheme implementation to
+    locate SLIB, load SLIB initialization file and so on.
+    EOS
+  end
```
Comment (mikemcquaid) [inline Library/Formula/slib.rb]: Don't need these caveats, thanks.

File: Library/Formula/slib.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+class Slib < Formula
+  homepage "http://people.csail.mit.edu/jaffer/SLIB"
+  url "http://groups.csail.mit.edu/mac/ftpdir/scm/slib-3b4.zip"
+  version "3b4"
```
Comment (mikemcquaid) [inline Library/Formula/slib.rb]: Is this a beta version?

File: Library/Formula/slib.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  def caveats; <<-EOS.undent
+    SLIB distribution is installed at /usr/local/lib/slib/.
+
+    See `info slib' to see how to configure your Scheme implementation to
+    locate SLIB, load SLIB initialization file and so on.
+    EOS
+  end
+
+  test do
+    system "#{bin}/slib", "-v"
```
Comment (mikemcquaid) [inline Library/Formula/slib.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

File: Library/Formula/slib.rb (hunk 2 above)
Comment (kmizumar) [inline Library/Formula/slib.rb]: SLIB version consists from 3 letters, major + minor + micro. and minor start from "a", then "b", then "c", and so on. I believe that this format is selected to make archive file name fit in 8+3 letters (for MS-DOS).

http://en.wikipedia.org/wiki/SLIB also says that 3b4 is stable release.

File: Library/Formula/slib.rb (hunk 1 above)
Comment (kmizumar) [inline Library/Formula/slib.rb]: I see. I'll delete it.

File: Library/Formula/slib.rb (hunk 3 above)
Comment (kmizumar) [inline Library/Formula/slib.rb]: Thank you so much for your comment. Would it be possible for you to let me know any examples which depend on other application formulae? To do meaningful test on SLIB, we need to know which one of the following scheme implementations is installed on the system: MIT scheme, scm, gsi, gosh, guile, slib48, larceny, scmlit, elk, sisc, kawa or mzscheme.

File: Library/Formula/slib.rb (hunk 2 above)
Comment (mikemcquaid) [inline Library/Formula/slib.rb]: Cool, thanks.

Comment (mikemcquaid) [inline Library/Formula/slib.rb]: Also: don't need to specify this as it's autodetected: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/19563/version=yosemite/testReport/junit/brew-test-bot/yosemite/audit_slib___strict/

File: Library/Formula/slib.rb (hunk 3 above)
Comment (mikemcquaid) [inline Library/Formula/slib.rb]: If this depends on a scheme implementation it should have a `depends_on` for that.

Comment (mikemcquaid) [inline Library/Formula/slib.rb]: And it doesn't need to be a particularly meaningful test; just run something more than `-v` or `-h` with the `slib` executable would be enough.

File: Library/Formula/slib.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
+class SchemeInstalled < Requirement
+  fatal true
+
```
Comment (mikemcquaid) [inline Library/Formula/slib.rb]: Can we add a `default_formula` here so it'll default to one of these if none are installed?

File: Library/Formula/slib.rb [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  depends_on SchemeInstalled
+
+  def install
+    system "./configure", "--prefix=#{prefix}"
+    system "make", "infoz", "INSTALL_INFO=install-info"
+    system "make", "install", "INSTALL_INFO=install-info"
+  end
+
+  test do
+    system "#{bin}/slib", "-v"
```
Comment (mikemcquaid) [inline Library/Formula/slib.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

File: Library/Formula/slib.rb (hunk 4 above)
Comment (kmizumar) [inline Library/Formula/slib.rb]: I personally prefer Gauche. Is it okay to put a line `default_formula "gauche"` in this formula?

File: Library/Formula/slib.rb (hunk 5 above)
Comment (kmizumar) [inline Library/Formula/slib.rb]: I tried to improve body of the test. Could you please check it in fbf4eee?

File: Library/Formula/slib.rb [hunk 6]
```diff
@@ -0,0 +1,84 @@
+class SchemeInstalled < Requirement
+  default_formula "gauche"
```
Comment (mikemcquaid) [inline Library/Formula/slib.rb]: :+1:

File: Library/Formula/slib.rb [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+    guile_bin = which("guile")
+    unless guile_bin.to_s == ""
+      ENV["GUILE_IMPLEMENTATION_PATH"] = Formula["guile"].prefix
+      ENV["GUILE_WARN_DEPRECATED"] = "no"
+
+      result = `#{guile_bin} -l ${SCHEME_LIBRARY_PATH}guile.init -l ${SCHEME_LIBRARY_PATH}guile.use test.scm`
+      assert_equal "slib", result
+      success = true
+    end
```
Comment (mikemcquaid) [inline Library/Formula/slib.rb]: Can just assume `gauche` is used in the `test do` block.

**Comment by DomT4** [comment]
What's the status here?

**Comment by DomT4** [comment]
Closing due to no response. Will welcome a new PR with the issues highlighted addressed. Thanks for your submission to Homebrew!

**Pull request closed without merging**